extensions/adapters/vector/src/test/java/mil/nga/giat/geowave/adapter/vector/AvroFeatureDataAdapterTest.java

```diff
@@ -0,0 +1,583 @@
+					"geometry:Geometry:srid=4326,pop:java.lang.Long,when:Date,whennot:Date,pid:String");// typeBuilder.buildFeatureType();
+
+			newFeature = FeatureDataUtils.buildFeature(
+					schema,
+					new Pair[] {
+						Pair.of(
+								"geometry",
+								factory.createPoint(new Coordinate(
+										27.25,
+										41.25))),
+						Pair.of(
+								"pop",
+								Long.valueOf(100)),
+						Pair.of(
+								"when",
+								time1),
+						Pair.of(
+								"whennot",
+								time2)
+
+					});
+		}
+		catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+
+	@Test
+	public void basicTest()
+			throws Exception {
+		final String password = "password";
+
+		final MiniAccumuloConfig config = new MiniAccumuloConfig(
+				Files.createTempDir(),
+				password);
+		config.setNumTservers(1);
+		MiniAccumuloCluster miniAccumulo = new MiniAccumuloCluster(
+				config);
+		miniAccumulo.start();
+
+		final String zookeeperUrl = miniAccumulo.getZooKeepers();
+		final String instancename = miniAccumulo.getInstanceName();
+		final String username = "root";
+		final String tableNamespace = "geowave";
+
+		final AccumuloOperations operations = new BasicAccumuloOperations(
+				zookeeperUrl,
+				instancename,
+				username,
+				password,
+				tableNamespace);
+
+		final AccumuloDataStore dataStore = new AccumuloDataStore(
+				operations);
+
+		final AvroFeatureDataAdapter adapter = new AvroFeatureDataAdapter(
+				schema);
+
+		final GeometryFactory geometryFactory = JTSFactoryFinder.getGeometryFactory();
+		final SimpleFeatureBuilder featureBuilder = new SimpleFeatureBuilder(
+				schema);
+		final Index index = IndexType.SPATIAL_VECTOR.createDefaultIndex();
+
+		final int numFeatures = 10;
+
+		// Write data using the whole feature data adapter
+		for (int i = 0; i < numFeatures; i++) {
+
+			final Point point = geometryFactory.createPoint(new Coordinate(
+					i,
+					i));
+
+			featureBuilder.set(
+					"geometry",
+					point);
+			featureBuilder.set(
+					"pop",
+					i);
+			featureBuilder.set(
+					"when",
+					new Date(
+							0));
+			featureBuilder.set(
+					"whennot",
+					new Date());
+
+			dataStore.ingest(
+					adapter,
+					index,
+					featureBuilder.buildFeature(Integer.toString(i)));
+		}
+
+		final Coordinate[] coordArray = new Coordinate[5];
+		coordArray[0] = new Coordinate(
+				-180,
+				-90);
+		coordArray[1] = new Coordinate(
+				180,
+				-90);
+		coordArray[2] = new Coordinate(
+				180,
+				90);
+		coordArray[3] = new Coordinate(
+				-180,
+				90);
+		coordArray[4] = new Coordinate(
+				-180,
+				-90);
+
+		// read data using the whole feature data adapter
+		final CloseableIterator<SimpleFeature> itr = dataStore.query(
+				adapter,
+				index,
+				new SpatialQuery(
+						new GeometryFactory().createPolygon(coordArray)));
+
+		int numReturned = 0;
+		while (itr.hasNext()) {
+			numReturned++;
+
+			final SimpleFeature feat = itr.next();
+
+			System.out.println("id: [" + feat.getID() + "]");
```

Comment:
This is a good debug test for a developer, but If we're going to use this as a unit test, we should probably remove prints to standard out and instead put in some assertions to validate the results of our query.